Industry-4-0-Design-Team/Industry_4.0_Website
# Frontpage navbar + gear additions

# Implementation description
Here are some additions I made:
- Added a gear icon that rotates with glow (size changes based on screen size to better fit the wide monitors + laptops)

![image](https://github.com/user-attachments/assets/d2f88f36-52ea-49c1-9f9b-e74c884f31d5)

- Added About Us and Mission sections

![image](https://github.com/user-attachments/assets/4d18da3d-b606-43e4-9de1-909722784598)


- Created a seperate component for the buttons
- Added a mobile sidebar nav that opens from a hamburger icon

![image](https://github.com/user-attachments/assets/355f9f0a-9d3d-473a-8b06-b9e5a19b1999)

- Changed nav placement so that the main section buttons are always in the middle, social media on the right, logo on the left (instead of them stretching out as the screen gets bigger it is kinda weird formatting that way)


the diamond background was removed (Tejas and I discussed this and we both agreed it would be easier to code and it does not look too bad this way anyways)


Steps to test
1. Hover over buttons and nav stuff (are these the way we want them?)
2. Fiddle around with the width on devtools idk (check mobile widths to test nav but remember that the page hasnt been formatted for mobile yet)
3. Click on "About" or "Sponsors" and see if this is how we want the page to automatically go there



# What should reviewers focus on?

- Is it intuitive that you can close the side navbar by clicking out of the navbar for mobile
- tailwind config: i sorta rawdogged all the code there is prolly many places where I could have used variables or created global variables for colours and text so maybe figure out what could be added?
- Do we like the hover styles? They were not on the figma so I kinda made up wahtever so maybe review those or keep an eye on those if we wanna change later
- Do yall fw the gear? is there anything here that needs changing?
- Do we want the nav to stick at the top like it is doing now. its also slightly blurred in the back which looks pretty clean but we can make it a solid colour if we want.


# Next steps

- Mobile responsiveness
- Add the remaining sections (come up with what text we want since the figma just has filler stuff there for now)
- Add all image assets (event pictures, team headshots, etc.)
- Page routing for seperate sponsors page
- Add footer
- Maybe have stuff fade in as we scroll down? Do whatever feels best
